### PR TITLE
CTCTRADERS-719: Auditing on submission

### PR DESCRIPTION
**Pull request by MatthewAlexanderBeck**

### CODE REVIEW

File: app/controllers/CheckYourAnswersController.scala [hunk 1]
```diff
@@ -79,15 +85,19 @@ class CheckYourAnswersController @Inject()(
         case Some(unloadingPermission) => {
           unloadingRemarksService.submit(arrivalId, request.userAnswers, unloadingPermission) flatMap {
             case Some(status) =>
+              auditEventSubmissionService.auditUnloadingRemarks(request.userAnswers, "submitUnloadingRemarks")
```
Comment (chris-ianson) [inline app/controllers/CheckYourAnswersController.scala:88]: Should the audit event go in the ACCEPTED block

File: app/navigation/Navigator.scala [hunk 2]
```diff
@@ -38,17 +38,15 @@ class Navigator @Inject()() {
     case CanSealsBeReadPage =>
       ua =>
         ua.get(CanSealsBeReadPage) match {
-          case Some(true)  => routes.AreAnySealsBrokenController.onPageLoad(ua.id, NormalMode)
-          case Some(false) => routes.AreAnySealsBrokenController.onPageLoad(ua.id, NormalMode)
-          case _           => routes.SessionExpiredController.onPageLoad() //TODO temporary redirect will be error page
+          case Some(_) => routes.AreAnySealsBrokenController.onPageLoad(ua.id, NormalMode)
```
Comment (chris-ianson) [inline app/navigation/Navigator.scala:41]: Good spot!

File: app/services/UnloadingRemarksService.scala [hunk 3]
```diff
@@ -70,18 +72,22 @@ class UnloadingRemarksService @Inject()(metaService: MetaService,
           None
       }
 
-  def resubmit(arrivalId: ArrivalId, eori: EoriNumber, userAnswers: UserAnswers)(implicit hc: HeaderCarrier): Future[Option[Int]] =
+  def resubmit(arrivalId: ArrivalId, userAnswers: UserAnswers)(implicit hc: HeaderCarrier): Future[Option[Int]] =
```
Comment (chris-ianson) [inline app/services/UnloadingRemarksService.scala:75]: Was the eori not used?

**Review by chris-ianson — CHANGES_REQUESTED** [review]
Couple of comments. Think we only audit if the submission is sent successfully.

File: app/controllers/CheckYourAnswersController.scala (hunk 1 above)
Comment (MatthewAlexanderBeck) [reply app/controllers/CheckYourAnswersController.scala:88]: Yes this makes sense as I guess we dont want to audit a failed submission

File: app/services/UnloadingRemarksService.scala (hunk 3 above)
Comment (MatthewAlexanderBeck) [reply app/services/UnloadingRemarksService.scala:75]: Yeah the param wasnt used so I did a little tidy up

Comment (chris-ianson) [reply app/services/UnloadingRemarksService.scala:74]: 👍

**Review by chris-ianson — APPROVED** [review]

**Pull request merged**